evproc/processor.py

```diff
@@ -308,11 +310,19 @@ def process(self, ev):
                   ``StopProcessing`` exception raised by a processor.
         """
 
-        # Walk through all the processors and process the event
-        for proc in self._get_procs(ev):
-            try:
-                proc(ev)
-            except exc.StopProcessing as ex:
-                return ex.retval
+        event_deque = deque()
+        event_deque.append(ev)
+        while len(event_deque):
```

Comment:
Shouldn't need the len() here; a simple "while event_deque" should do the trick.